### PR TITLE
[MacOS] Enable game mode when switching to full screen mode

### PR DESCRIPTION
**Pull request by alphavector**
Related  #1515

I haven't tested how much performance has improved, need help with before/after performance tests

Screenshot 2024-11-17 at 04 00 05

Screenshot 2024-11-17 at 04 00 19

### CODE REVIEW

**Comment by alphavector** [comment]
The game mode also extends to the game launcher. "Ideally", game mode should only apply to the game, but that doesnt seem to be a problem at the moment

File: dist/MacOSBundleInfo.plist.in [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	<key>CFBundleExecutable</key>
+	<string>shadps4</string>
+
+	<key>CFBundleVersion</key>
+	<string>1.0.0</string>
+	<key>CFBundleShortVersionString</key>
+	<string>${MACOSX_BUNDLE_SHORT_VERSION_STRING}</string>
+
+	<key>LSMinimumSystemVersion</key>
+	<string>${CMAKE_OSX_DEPLOYMENT_TARGET}</string>
```
Comment (squidbus) [inline dist/MacOSBundleInfo.plist.in:23]: Have you verified this results in the correct value? I remember running into problems in the past where only certain variables are recognized in cmake templated plists.

Comment (alphavector) [reply dist/MacOSBundleInfo.plist.in:23]: yes, this leads to the correct value

```sh
> grep 'LSMinimumSystemVersion' -A 1 build/shadps4.app/Contents/Info.plist
        <key>LSMinimumSystemVersion</key>
        <string>14</string>
```

**Comment by squidbus** [comment]
Need to add `dist/MacOSBundleInfo.plist.in` to the `REUSE.toml`, otherwise should be good.

**Comment by ghost** [comment]
I tested USF 4 because of the input lag and Ultimate Marvel vs Capcom 3 because of the partly stuttering. Technically it works but using the game mode didn't make a difference, neither visible nor felt. Also fps is the same. Wild guess: Could either be that the emulation for these two games is currently not as good as i could be andI therefore couldn't see a difference or the game mode may has no impact when Rosetta 2 is used.

**Comment by squidbus** [comment]
I honestly don't expect game mode to have much of a difference on game performance in general.

**Comment by ghost** [comment]
Yeah. Might be there's a difference when playing COD with the rpcs3 arm64 version ... will check that.

**Pull request merged**